### ISSUE TITLE
Feature request: integrate `crev` into Pkg

### ISSUE THREAD

**Issue by DilumAluthge** [JuliaLang/Pkg.jl#1858]
This is a meta-issue for the integration of `crev` into Pkg.

We don't currently have an issue to track this, which is why I'm opening this.

I think this is a different issue from https://github.com/JuliaLang/Pkg.jl/issues/1856. In particular, this issue is to specifically track the integration of `crev`, as opposed to another system.

Related discussions:
- https://github.com/JuliaComputing/JuliaHub/issues/2

### Prior Art

- https://github.com/crev-dev/cargo-crev

**Comment by tkf** [comment]
I don't know how crev works exactly yet to understand how reasonable it is to use it only on the server side.

But from a "social" aspect, I wonder if hiding crev from end-users somewhat impedes submitting reviews on packages.  IIUC, a nice thing in Pkg2 (which was inherited to Pkg3) was that it's so easy to turn a user to a contributor because the repo is right there in `~/.julia`.  Similarly, if we have crev frontend that is easy to use for validation as well as submitting a review, it may help us build the web of trust.  Of course, this could just be a matter of UI.

**Comment by non-Jedi** [comment]
@StefanKarpinski is the idea then that the server chooses a set of reviewers to trust, and you lose the individualized ability to choose whose review to trust?

**Comment by StefanKarpinski** [comment]
No, the idea is that the metadata publisher checks crev input, verifies it and then publishes who signed what without all the cryptographic proof. The client can still use any policy they want, but the they don't get or review the cryptographic signatures (which they're not well-positioned to do anyway). If you want to review the crev claims yourself, you can do so separately.

**Comment by KristofferC** [comment]
I think this should be done as a separate tool and if/when it exists we can add some kind of more official plugin mechanics to pkg to use it.
